bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -1121,11 +1134,28 @@ public void run()
             }
           }
         }
-      } catch (Exception e) {
-        logger.error("{}", DataList.this, e);
+      } catch (RuntimeException e) {
```

Comment:
Do we know scenarios where an exception will be thrown in the above block or is this more a safety precaution?